packages/hopli/src/network_registry.rs

```diff
@@ -0,0 +1,71 @@
+use crate::process::{child_process_call_foundry_self_register, set_process_path_env};
+use clap::Parser;
+
+use crate::utils::{Cmd, HelperErrors};
+
+/// CLI arguments for `hopli network-registry`
+#[derive(Parser, Default, Debug)]
+pub struct NetworkRegistryArgs {
+    #[clap(help = "Environment name. E.g. monte_rosa", long)]
+    environment_name: String,
+
+    #[clap(help = "Environment type. E.g. production", long, short)]
+    environment_type: String,
+
+    #[clap(
+        help = "Comma sperated node peer ids",
+        long,
+        short,
+        default_value = None
+    )]
+    peer_ids: String,
+
+    #[clap(
+        help = "Specify path pointing to the foundry root",
+        long,
+        short,
+        default_value = None
+    )]
+    contracts_root: Option<String>,
+
+    #[clap(
+        help = "Private key of the caller address, e.g. 0xabc",
```

Comment:
The secret should not be passed as a parameter. Either as an environment variable or a path to a file which contains the password.

Reply:
Updated. `password` can be an env variable or pass via file, and `private_key` should be an env variable